### ISSUE TITLE
Use CI to automatically build an up-to-date paper.

### ISSUE THREAD

**Issue by roipoussiere** [DemocracyEarth/paper#210]
Using a Continuous Integration server could add interesting features which improve the project:

- **transparency**: the user will have the possibility to build the PDF by himself: in this way it will be a proof that the PDF is the actual paper written collectively.
- **accessibility**: providing the paper in e-pub makes it easier to read.
- **accessibility**: all the translations of the paper can be downloaded in PDF and e-pub.

The first purpose here is to have the [PDF file](https://github.com/DemocracyEarth/paper/blob/master/The%20Social%20Smart%20Contract.pdf) always up-to-date with the [source](https://github.com/DemocracyEarth/paper/blob/master/README.mediawiki), and don't worry about building the pdf file, which is an useless spent time.
The server can for example updates a gh-pages website which include a pretty-printed only version of the paper, plus a link to the PDF of the paper (and why not other formats).
We can for example use [Travis](https://travis-ci.com/), which is free for open-source projects (yay!).

I actually think about this workflow:
- someone merge or push something on master;
- Travis is triggered and download the content;
- with [Pandoc](http://pandoc.org/), Travis builds the PDF and the e-book;
- Travis creates a website with for example [Jekyll](https://jekyllrb.com/) or [mkdocs](http://www.mkdocs.org/);
- Travis push all of this to gh-pages and every one access it fews minutes after the push/merge.

I have a little experience in continuous integration, and if you are agree with this issue, I can see you a proof of concept of this this week-end.

**Comment by domi41** [comment]
Note that the bibliography is not available in the mediawiki file, it's only in the PDF.

**Comment by roipoussiere** [comment]
> Note that the bibliography is not available in the mediawiki file, it's only in the PDF.

Indeed! Few resources (such as the cover, LaTeX sources and bib) must be shared to do this.

**Comment by domi41** [comment]
The PDF is usually committed by @santisiri ; @santisiri can you commit the sources for PDF generation as well ?

**Comment by santisiri** [comment]
i think this is a great idea.

the PDF was done manually for aesthetic reasons, but a mediawiki to LaTeX importer could be easily implemented. there's a bunch of projects for that on github, some of them run on python. I haven't found time to implement them, but if someone wants to make that happen, it would be great!

also, happy to accept any pull request regarding CI, i think it's a great idea.

**Comment by domi41** [comment]
@santisiri How strongly do you feel about mediawiki ? We find its syntax cumbersome and not literature-friendly (nor human-friendly for that matter). A markdown to LaTeX should be easy enough to implement too. It might require some tweaking to handle the "paper" format (cover, bib, quotes, alignments), but nothing too complicated. Might even make a useful lib for others.

@roipoussiere Would love to see a PoC of this as well !

**Comment by santisiri** [comment]
@domi41 what pushed me to mediawiki was that 

- it rendered properly on github
- automatically generates index
- it's used by @bitcoin core's BIP process  
- it's wiki friendly

that said, i'm all in favor for including a latex version in this repo (and even a markup .md one).
